utility/src/main/java/org/oppia/android/util/profile/ProfileNameValidator.kt

```diff
@@ -0,0 +1,55 @@
+package org.oppia.android.util.profile
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Utility to validate that profile names are correctly formatted. */
+@Singleton
+class ProfileNameValidator @Inject constructor() {
+  private val noRepeatedAllowedSymbolsRegex by lazy { Regex("[.'-]{2}") }
+
+  /**
+   * Validates a profile name to ensure that it isn't confusingly formatted
+   * or contains invalid characters.
+   *
+   * @param name name of the profile
+   * @return if the profile name is allowed or not
+   */
+  fun isNameValid(name: String): Boolean {
+    return (onlyLettersAndAllowedSymbols(name) && noRepeatedUseOfAllowedSymbols(name))
+  }
+
+  /** Validates if the character in the name is an alphabet or an allowed symbol or not. */
+  private fun onlyLettersAndAllowedSymbols(name: String): Boolean {
+    name.forEach {
+      if (!(it.isAlphabetic() || isSymbolAllowed(it))) {
+        return false
+      }
+    }
+    return true
+  }
+
+  private fun noRepeatedUseOfAllowedSymbols(name: String): Boolean {
+    return !name.contains(noRepeatedAllowedSymbolsRegex)
+  }
+
+  private fun isSymbolAllowed(symbol: Char): Boolean {
+    return symbol == '.' || symbol == '-' || symbol == '\''
+  }
+
+  private fun Char.isAlphabetic(): Boolean {
+    /**
```

Comment:
Should be a block comment, or multiple line comments. KDocs should only be reserved for top-level members like classes, objects, functions, and values.

Reply:
Noted. Done.